internal/services/securitycenter/security_center_server_vulnerability_assessment_hybrid_vm_resource.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		if !utils.ResponseWasNotFound(vulnerabilityAssessment.Response) {
+			return fmt.Errorf("checking for presence of existing Advanced Threat Protection for %s: %+v", *hybridMachineId, err)
+		}
+	}
+
+	if vulnerabilityAssessment.ID != nil && *vulnerabilityAssessment.ID != "" {
+		return tf.ImportAsExistsError("azurerm_security_center_server_vulnerability_assessment_hybrid_vm", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %s: %+v", *hybridMachineId, err)
+	}
+
+	if vulnerabilityAssessment.ID != nil {
+		d.SetId(*vulnerabilityAssessment.ID)
+	}
+
+	// Take some time for resource to be visible after creating. Hence the retry logic
+	readError := resourceServerVulnerabilityAssessmentRead(d, meta)
+	for retry := 1; d.Id() == "" && retry < 11; retry++ {
+		time.Sleep(time.Duration(retry) * time.Second)
+		readError = resourceServerVulnerabilityAssessmentRead(d, meta)
+	}
+	return readError
+}
+
+func resourceServerVulnerabilityAssessmentHybridVMRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	hybridMachineId, err := computeParse.HybridMachineID(d.Get("hybrid_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("Server Vulnerability Assessment was not found for %s: %+v", *hybridMachineId, err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Server Vulnerability Assessment status for %s: %+v", *hybridMachineId, err)
+	}
+
+	d.SetId(*resp.ID)
+	d.Set("hybrid_machine_id", hybridMachineId.ID())
+
+	return nil
+}
+
+func resourceServerVulnerabilityAssessmentHybridVMDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	hybridMachineId, err := computeParse.HybridMachineID(d.Get("hybrid_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	// Can not delete if still in provisioning state. Wait for it to complete.
+	for retry := 1; retry <= 20; retry++ {
+		response, err := client.Get(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+		if err != nil || response.ProvisioningState != security.ProvisioningState1Provisioning {
+			break
+		}
+		time.Sleep(time.Duration(retry) * time.Second)
+	}
```

Comment:
this would indicate that it's not fully provisioned in the Create, can we poll there instead?